splashsurf_lib/src/mesh.rs

```diff
@@ -13,6 +13,15 @@ pub struct TriMesh3d<R: Real> {
     pub triangles: Vec<[usize; 3]>,
 }
 
+impl<R: Real> Default for TriMesh3d<R> {
+    fn default() -> Self {
+        Self {
+            vertices: Vec::new(),
+            triangles: Vec::new(),
+        }
+    }
+}
+
```

Comment:
I think you can just derive this? Then I would also add this derive for all other mesh types, to be consistent.

Reply:
so if i derive the default then i need a default bound in many other places; it seems less desirable. For example:

```
   --> splashsurf_lib/src/lib.rs:219:23
    |
219 |     let mut surface = SurfaceReconstruction::default();
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `R`
    |
    = note: required because of the requirements on the impl of `Default` for `SurfaceReconstruction<_, R>`
    = note: required by `std::default::Default::default`
help: consider further restricting this bound
    |
214 | pub fn reconstruct_surface<I: Index, R: Real + Default>(
```

I can add this if you want but i think the default bound is kind of spurious and not helpful?

Reply:
Huh, I didn't expect this, as the `Default` impl of `Vec` doesn't require `T` (`R` in this case) to implement `Default`... But I think it's some general issue with derive, see https://github.com/rust-lang/rust/issues/26925. It adds a trait bound to all generic arguments even if this isn't actually needed for the derive.

To still use the derive approach, we could simply add `Default` as a trait bound to the blanket implementation of the `Real` trait. This wouldn't lead to changes anywhere else in the code. I think this would be fair because nearly all numeric types implement `Default` as returning the zero value. This would clean the code up a bit because we would not need all these manual implementations. But I can do this myself later.

Reply:
Btw. I think for the other types where you added the Default derive, it would actually result in the same errors if it was used in the library...